### DISCUSSION TITLE
How do I use `vcpkg_cmake_config_fixup()` with multiple `*-config.cmake` files

### DISCUSSION THREAD

**Discussion by coryan** [microsoft/vcpkg#20511] [Q&A] [answered]
I have a port (`google-cloud-cpp`) that creates multiple `*-config.cmake` files, each intended for different `find_package()` calls (for example `google_cloud_cpp_storage` and `google_cloud_cpp_bigtable`).

The bots are recommended that I update the port to `vcpkg_cmake_config_fixup()`. I cannot figure out how to use that function in my case.  It installs all the config files in (for example) `share/google-cloud-cpp/google_cloud_cpp_bigtable/google_cloud_cpp_bigtable-config.cmake` whereas I think I just want `share/google_cloud_cpp_bigtable/google_cloud_cpp_bigtable-config.cmake`.

Is there a flag or something else I am missing?

**Comment by Osyotr** [comment, marked as answer]
I think the right way to do it is like so:
```cmake
vcpkg_cmake_config_fixup(PACKAGE_NAME google_cloud_cpp_storage CONFIG_PATH lib/cmake/google_cloud_cpp_storage DO_NOT_DELETE_PARENT_CONFIG_PATH)
vcpkg_cmake_config_fixup(PACKAGE_NAME google_cloud_cpp_bigtable CONFIG_PATH lib/cmake/google_cloud_cpp_bigtable DO_NOT_DELETE_PARENT_CONFIG_PATH)
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake" "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")
```
You probably can omit last line together with `DO_NOT_DELETE_PARENT_CONFIG_PATH`.

**Reply by coryan** [reply]
Thanks, I think that is it.  A bit tedious (#20521) but doable.
